src/tokens/TokenBalances.tsx

```diff
@@ -127,6 +146,18 @@ function TokenBalancesScreen({ navigation }: Props) {
     })
   }
 
+  const handleMeasureBalanceHeight = (event: LayoutChangeEvent) => {
+    balanceHeightRef.current = event.nativeEvent.layout.height
+  }
+
+  const debouncedUpdateHeader = debounce((scrollOffset: number) => {
+    setShowBalanceInHeader(scrollOffset > balanceHeightRef.current)
+  }, HEADER_UPDATE_DEBOUNCE_TIME_MS)
+
+  const handleScroll = (event: { nativeEvent: NativeScrollEvent }) => {
+    debouncedUpdateHeader(event.nativeEvent.contentOffset.y)
```

Comment:
Would be better to use an `Animated.Value` and update the opacity of the header, after a certain threshold.
This way no need to debounce and no arbitrary delay.